js/stringProcessing/urlStructure.js

```diff
@@ -0,0 +1,114 @@
+let core = require( "tokenizer2/core" );
+
+var urlTokenizer;
+let tokens;
+
+let staticRegex = /^[^%]+$/;
+let variableRegex = /^%%[^%]+%%$/;
+
+/**
+ * Creates a tokenizer to tokenize HTML into blocks.
+ *
+ * @returns {void}
+ */
+function createTokenizer() {
+	tokens = [];
+
+	urlTokenizer = core( function( token ) {
+		tokens.push( token );
+	} );
+
+	urlTokenizer.addRule( staticRegex, "static" );
+	urlTokenizer.addRule( variableRegex, "variable" );
+}
+
+
+/**
+ * Represents a URL structure. Placeholders can be defined using %%placeholder%% and can later be filled using the `applyData` method.
+ */
+class UrlStructure {
+
+	/**
+	 * Sets the structure to the passed structure.
+	 *
+	 * @param {Array} structure The structure of the URL.
+	 */
+	constructor( structure ) {
+		this._structure = structure;
```

Comment:
You're using __structure as a class variable, shouldn't this be defined as a class variable before setting it in the constructor?